staging/src/k8s.io/kubectl/pkg/cmd/patch/patch.go

```diff
@@ -189,6 +189,11 @@ func (o *PatchOptions) Validate() error {
 	if o.Local && o.dryRunStrategy == cmdutil.DryRunServer {
 		return fmt.Errorf("cannot specify --local and --dry-run=server - did you mean --dry-run=client?")
 	}
+	if !o.Local && cmdutil.IsFilenameSliceEmpty(o.FilenameOptions.Filenames, o.FilenameOptions.Kustomize) {
```

Comment:
That error is coming from https://github.com/kubernetes/kubernetes/blob/dcec526c805e371eebfc9cee8835df7f542cfa93/staging/src/k8s.io/cli-runtime/pkg/resource/builder.go#L842 and since that's library code it should not mention any flags at all directly. I'd suggest reviewing error messages in that file and fixing it accordingly there.